.github/workflows/tiledb-go.yml

```diff
@@ -123,7 +123,6 @@ jobs:
 
   Linux_Address_Sanitizer:
     # Only run this job for releases
-    if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
Just testing with CI on this PR, once I see it's passing I'll revert this so it's only ran for releases.

Reply:
Reverted this, but kept the removal of `continue-on-error`, there is a full CI run here with passing ASAN jobs

https://github.com/TileDB-Inc/TileDB-Go/actions/runs/10497212377/job/29079527674?pr=345